src/baldr/edge_elevation.cc

```diff
@@ -0,0 +1,53 @@
+#include <cmath>
+#include <valhalla/midgard/logging.h>
+#include "baldr/edge_elevation.h"
+
+namespace valhalla {
+namespace baldr {
+
+// Constructor with arguments.
+EdgeElevation:: EdgeElevation(const float mean_elev, const float max_up_slope,
+                              const float max_down_slope) {
+  set_mean_elevation(mean_elev);
+  set_max_up_slope(max_up_slope);
+  set_max_down_slope(max_down_slope);
+}
+
+// Set the mean elevation.
+void EdgeElevation::set_mean_elevation(const float mean_elev) {
```

Comment:
will look into this. GraphTile currently returns a pointer (or nullptr if the tile does not have elevation). Maybe have a static NoData EdgeElevation struct that gets pointed to in these cases.